packages/react/src/utils/polymorphic-type/index.ts

```diff
@@ -51,7 +51,7 @@ interface ForwardRefComponent<
   <As = IntrinsicElementString>(
     props: As extends ''
       ? { as: keyof JSX.IntrinsicElements }
-      : As extends React.ComponentType<infer P>
+      : As extends React.ComponentType<React.PropsWithChildren<infer P>>
```

Comment:
Is this a safe assumption to make? I don't know that the polymorphic type will always be on a component with children.

Reply:
This was done by the codemod (not by hand). I'll see what happens when I revert this change.